main.js

```diff
@@ -0,0 +1,122 @@
+//Fix problem with deleting comments and adding new comments
+
+//Fix problem with comments section showing comments for every post no matter what post was clicked
+
+
+
+//create list of posts and names to be associated with those posts
+var postsArray = [];
+
+//adding each post to list of posts
+var postsObj = function(msg, name) {
+  postsArray.push({
+    userMsg: msg,
+    userName: name,
+    post: postsArray.length + 1,
+    comments: []
+  })
+};
+
+//Allows user to post upon clicking 'post'
+$('#post').on('click', function() {
+  postsObj($('#message').val(), $('#name').val());
+  console.log(postsArray[postsArray.length - 1].post);
+  //console.log(postsArray);
+  var newPost = '<div><div><button type="button" style="padding:0px" class="btn-link remove">remove</button><button type="button" class="btn-link comments" id="' + postsArray[postsArray.length - 1].post + '" data-toggle="modal" data-target="commentsModal">comments</button></div><div class="msg">' + postsArray[postsArray.length - 1].userMsg + '</div><div class="name">Posted by: <b>' + postsArray[postsArray.length - 1].userName + '</b></div><hr></div>'
+  $('.posts').append(newPost);
```

Comment:
If you add the post directly to the DOM from here, what was the point in creating the `postsArray`?